Modules/Salary/Resources/assets/js/components/SalaryBreakdown.vue

```diff
@@ -84,8 +201,78 @@ export default {
 		totalSalary() {
 			return this.basicSalary + this.hra + this.transportAllowance + this.foodAllowance + this.otherAllowance;
 		},
+		employeeEsi() {
+			if(this.grossSalary < this.salaryConfigs.employee_esi_limit.fixed_amount) {
+				let percentage = this.salaryConfigs.employee_esi.percentage_rate;
+				return Math.ceil(this.grossSalary * percentage / 100);
+			}
+			return 0;
+		},
+		employeeEpf() {
+			let multiplier = this.grossSalary;
+			if (this.salaryConfigs.employee_epf.percentage_applied_on == "basic_salary") {
+				multiplier = this.basicSalary;
+			}
+			let percentage = parseInt(this.salaryConfigs.employee_epf.percentage_rate);
+			return Math.ceil(multiplier * percentage / 100);
+		},
+		totalDeduction() {
+			return this.employeeEsi + this.employeeEpf + this.foodAllowance;
+		},
+		netPay() {
+			return this.totalSalary - this.totalDeduction;		
+		},
+		employerEsi() {
+			if(this.grossSalary < this.salaryConfigs.employer_esi_limit.fixed_amount) {
+				let percentage = this.salaryConfigs.employer_esi.percentage_rate;
+				return Math.ceil(this.grossSalary * percentage / 100);
+			}
+			return 0;
+		},
+		employerEpf() {
+			let multiplier = this.grossSalary;
+			if (this.salaryConfigs.employer_epf.percentage_applied_on == "basic_salary") {
+				multiplier = this.basicSalary;
+			}
+			let percentage = parseInt(this.salaryConfigs.employer_epf.percentage_rate);
+			return Math.ceil(multiplier * percentage / 100);
+		},
+		administrationCharges() {
+			let multiplier = this.grossSalary;
+			if (this.salaryConfigs.administration_charges.percentage_applied_on == "basic_salary") {
+				multiplier = this.basicSalary;
+			}
+			let percentage = this.salaryConfigs.administration_charges.percentage_rate;
+			return Math.ceil(multiplier * percentage / 100);
+		},
+		edliCharges() {
+			let multiplier = this.grossSalary;
+			if (this.salaryConfigs.edli_charges.percentage_applied_on == "basic_salary") {
+				multiplier = this.basicSalary;
+			}
+			let percentage = this.salaryConfigs.edli_charges.percentage_rate;
+			return Math.min(Math.ceil(multiplier * percentage / 100) , Math.ceil(this.salaryConfigs.edli_charges_limit.fixed_amount * percentage / 100));
+		},
+		ctc() {
+			return Math.ceil(parseInt(this.grossSalary) + parseInt(this.employerEsi) + parseInt(this.employerEpf) + parseInt(this.administrationCharges) + parseInt(this.edliCharges));
+		},
+		ctcAnnual() {
+			return this.ctc * 12;		
+		},
+		healthInsurance() {
+			if (this.grossSalary === "") {
+				return 0;
+			}
+			if(this.employerEsi === null || this.emplpyeeEsi === null)
+			{
+				return parseInt(this.salaryConfigs.health_insurance.fixed_amount);
+			}
+			return 0;
```

Comment:
I think we can combine
```suggestion
			if (this.grossSalary === "" || this.employerEsi === null || this.emplpyeeEsi === null) {
				return 0;
			}
			return parseInt(this.salaryConfigs.health_insurance.fixed_amount);
```

Reply:
ok